assets/js/app/Data.js

```diff
@@ -47,9 +99,19 @@ LocusZoom.Data.Requester = function(sources) {
     };
 };
 
-LocusZoom.Data.AssociationSource = function(url) {
-    this.url = url;
-
+LocusZoom.Data.AssociationSource = function(init) {
+    if (LocusZoom.isString(init)) {
+        this.url = init;
+        this.params = {};
+    } else {
+        this.url = init.url;
+        this.params = init.params || {};
+    }
+    if (!this.url) {
+        throw("AssociationSource not initialized with required URL");
+    }
+    
+    var that=this;
```

Comment:
If you're not a huge fan of cloning `this` into another var (`that`) to have as a reference in a scoped function then `.bind()` can help. This of course only works when the scoped function isn't already referring to itself with `this`, which is the case for `this.getData()`. The syntax would go something like:

```javascript
this.getData = function(state, fields, outnames) {
    // stuff, including references to "this"
}).bind(this);
```

It's arbitrary which we use. My reasons for favoring `bind()` are that `this` is a known special keyword and intuitively when working with a function's logic I expect `this` to refer to an object or thing that's a major part of my context. If that same context comes from another variable it works the same, but I need to keep that other variable's name in mind, which can just be a bit less intuitive.